### ISSUE TITLE
Cannot access Discourse if user have changed username / email associated to their edx account

### ISSUE THREAD

**Issue by brospars** [mit-teaching-systems-lab/discourse-edx-lti#32]
If an user changes his username or email associated with their edx account, then he cannot access Discourse because of [this](https://github.com/mit-teaching-systems-lab/discourse-edx-lti/blob/master/lti_authenticator.rb#L66).

Wouldn't it be possible to update his account instead of throwing an error ?

**Comment by kevinrobinson** [comment]
@brospars Thanks!  In my limited experience, this hasn't been an issue, so that's why it's more conservative now.  But that may be related to the particular courses, or that they've been smaller and we just haven't run into anyone changing their name yet. :)

If you want to submit a pull request changing this, please include links to EdX documentation and Discourse docs about this, as the plugin would have to make the same assumptions about which is the primary key and how changes are handled (eg, as in https://support.edx.org/hc/en-us/articles/115016004448-Can-I-change-my-edX-username- for EdX).

**Comment by brospars** [comment]
Yes Edx use the username as primary key, so this plugin could make the same assumption and if the email has changed then update it.

I'm leaving for some days but I will look into it next week

**Comment by kevinrobinson** [comment]
@brospars Sounds good, pull requests welcome!  👍 

I'm also happy to brainstorm on how to verify this is works correctly for the various edge cases, since there's a bunch of risk in deploying that I likely won't be able to verify myself.

**Comment by brospars** [comment]
@kevinrobinson Hello just to let you know I've opened two PR would you mind taking a look ?
